test/OpenTelemetry.Tests.Stress/README.md

```diff
@@ -73,59 +92,88 @@ process_runtime_dotnet_gc_allocations_size_bytes 5485192 1658950184752
 
-public partial class Program
+public static class Program
 {
-    public static void Main()
+    public static int Main(string[] args)
     {
-        Stress(concurrency: 10, prometheusPort: 9464);
+        return StressTestFactory.RunSynchronously<MyStressTest>(args);
     }
 
-    [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    protected static void Run()
+    private sealed class MyStressTest : StressTest<StressTestOptions>
     {
-        // add your logic here
+        public MyStressTest(StressTestOptions options)
+            : base(options)
+        {
+        }
+
+        protected override void RunWorkItemInParallel()
+        {
+        }
     }
 }
 ```
 
-Add the Skeleton.cs file to your `*.csproj` file:
+Add the following project reference to the project:
 
 ```xml
-  <ItemGroup>
-    <Compile Include="Skeleton.cs" />
-  </ItemGroup>
+<ProjectReference Include="$(RepoRoot)\test\OpenTelemetry.Tests.Stress\OpenTelemetry.Tests.Stress.csproj" />
 ```
 
-Add the following packages to the project:
+Now you are ready to run your own stress test. Add test logic in the
+`RunWorkItemInParallel` method to measure performance.
 
-```shell
-dotnet add package OpenTelemetry.Exporter.Prometheus --prerelease
-dotnet add package OpenTelemetry.Instrumentation.Runtime --prerelease
-```
+To define custom options create an options class which derives from
+`StressTestOptions`:
 
-Now you are ready to run your own stress test.
+```csharp
+using CommandLine;
+using OpenTelemetry.Tests.Stress;
+
+public static class Program
+{
+    public static int Main(string[] args)
+    {
+        return StressTestFactory.RunSynchronously<MyStressTest, MyStressTestOptions>(args);
+    }
+
+    private sealed class MyStressTest : StressTest<MyStressTestOptions>
+    {
+        public MyStressTest(MyStressTestOptions options)
+            : base(options)
+        {
+        }
+
+        protected override void RunWorkItemInParallel()
+        {
+            // Use this.Options here to access options supplied
+            // on the command line.
+        }
+    }
+
+    private sealed class MyStressTestOptions : StressTestOptions
+    {
+        [Option('r', "rate", HelpText = "Add help text here for the rate option. Default value: 0.", Required = false)]
+        public int Rate { get; set; } = 0;
+    }
+}
+```
 
 Some useful notes:
 
-* You can specify the concurrency using `Stress(concurrency: {concurrency
-  number})`, the default value is the number of CPU cores. Keep in mind that
-  concurrency level does not equal to the number of threads.
-* You can specify a local PrometheusExporter listening port using
-  `Stress(prometheusPort: {port number})`, the default value is `0`, which will
-  turn off the PrometheusExporter.
-* You want to put `[MethodImpl(MethodImplOptions.AggressiveInlining)]` on
-  `Run()`, this helps to reduce extra flushes on the CPU instruction cache.
 * You might want to run the stress test under `Release` mode rather than `Debug`
```

Comment:
Change this to SHOULD?

Reply:
I reworded it to make it stronger and mentioned the warning message we write to the console in `Debug`.